### PR TITLE
add api controller

### PR DESCRIPTION
**Pull request by arufanov**
Add api controller

### CODE REVIEW

**Comment by arufanov** [comment]
@yurichervoniy lets discuss:
*where is api tests

**Review by yurichervoniy — APPROVED** [review]

File: src/main/java/com/a5lab/tabr/controllers/TenantsApiController.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import com.a5lab.tabr.domain.tenants.Tenant;
+import com.a5lab.tabr.domain.tenants.TenantService;
+
+@RestController
+@RequiredArgsConstructor
+public class TenantsApiController {
+
+  private final TenantService tenantService;
+
+  @GetMapping("/api/tenants")
```
Comment (yurichervoniy) [inline src/main/java/com/a5lab/tabr/controllers/TenantsApiController.java:22]: api/v1/ - base path

File: src/main/java/com/a5lab/tabr/controllers/TenantsApiController.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  public List<Tenant> index() {
```
Comment (yurichervoniy) [inline src/main/java/com/a5lab/tabr/controllers/TenantsApiController.java:23]: It's better to return page or slice

File: src/main/java/com/a5lab/tabr/domain/tenants/TenantService.java [hunk 3]
```diff
@@ -1,12 +1,15 @@
...
 
+import java.util.List;
 import java.util.Optional;
 
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
 
 public interface TenantService {
 
+  List<Tenant> findAll();
```
Comment (yurichervoniy) [inline src/main/java/com/a5lab/tabr/domain/tenants/TenantService.java:11]: I prefer to return Collection<Tenant> or Iterable<Tenant> - i think you don't need List specification, for RandomAcces by position and so on. 
Advice - use more abstract type

File: src/main/java/com/a5lab/tabr/domain/tenants/TenantServiceImpl.java [hunk 4]
```diff
@@ -15,6 +16,13 @@ public class TenantServiceImpl implements TenantService {
   private final TenantRepository tenantRepository;
   private final TenantMapper tenantMapper;
 
+  @Override
+  @Transactional(readOnly = true)
+  public List<Tenant> findAll() {
```
Comment (yurichervoniy) [inline src/main/java/com/a5lab/tabr/domain/tenants/TenantServiceImpl.java:21]: Return more Abstract type. I dont think that you need exatcly List

File: src/main/java/com/a5lab/tabr/controllers/TenantsApiController.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.a5lab.tabr.domain.tenants.Tenant;
+import com.a5lab.tabr.domain.tenants.TenantService;
+
+@RestController
+@RequiredArgsConstructor
+public class TenantsApiController {
```
Comment (yurichervoniy) [inline src/main/java/com/a5lab/tabr/controllers/TenantsApiController.java:18]: Usually on top of controller we write base path 
@RequestMapping(path="api/v1/tenants) 
All methods below inheritece base controller path

File: src/main/java/com/a5lab/tabr/controllers/TenantsApiController.java (hunk 1 above)
Comment (arufanov) [reply src/main/java/com/a5lab/tabr/controllers/TenantsApiController.java:22]: @yurichervoniy 
i have added @RequestMapping("/api/v1") for controller, but not sure that is a best way. I wish to do it for all controllers: api/v1, but don't understand how to distinguish web & rest api.

File: src/main/java/com/a5lab/tabr/domain/tenants/TenantService.java (hunk 3 above)
Comment (arufanov) [reply src/main/java/com/a5lab/tabr/domain/tenants/TenantService.java:11]: @yurichervoniy 
* changed to collection. done

File: src/main/java/com/a5lab/tabr/domain/tenants/TenantServiceImpl.java (hunk 4 above)
Comment (arufanov) [reply src/main/java/com/a5lab/tabr/domain/tenants/TenantServiceImpl.java:21]: @yurichervoniy 
changed to collection. done

File: src/main/java/com/a5lab/tabr/controllers/TenantsApiController.java (hunk 5 above)
Comment (arufanov) [reply src/main/java/com/a5lab/tabr/controllers/TenantsApiController.java:18]: @arufanov 
I have added:  @RequestMapping("/api/v1"), but see prev comment: "i have added @RequestMapping("/api/v1") for controller, but not sure that is a best way. I wish to do it for all controllers: api/v1, but don't understand how to distinguish web & rest api. I need something like this "server.servlet.context-path=/api/v1" for rest controller and "server.servlet.context-path=/1" for web controllers"

File: src/main/java/com/a5lab/tabr/controllers/TenantsApiController.java (hunk 2 above)
Comment (arufanov) [reply src/main/java/com/a5lab/tabr/controllers/TenantsApiController.java:23]: @yurichervoniy 
* create independent issue: https://github.com/a5lab/tabr/issues/130. Could you provide example based on you experience. Thanks!

**Review by yurichervoniy — APPROVED** [review]

**Review by yurichervoniy — APPROVED** [review]

**Pull request merged**